.vscode/settings.json

```diff
@@ -1,8 +1,11 @@
 {
   "editor.detectIndentation": true,
-  "editor.formatOnSave": true,
+  "editor.formatOnSave": false,
```

Comment:
Best to make sure that changes in the PR are relevant to the change being made. Worth going through and undoing any irrelevant changes like this change here.

Reply:
Apologies. i didn't want to do any changes on the .vscode environment. I just checked the git diff for the relevant content. Somehow i noticed that the auto-formatting was not accordingly to the old code, so i disabled it. but revert back.

Reply:
> Somehow i noticed that the auto-formatting was not accordingly to the old code

That was a good call. After this PR we should do a no logic changes style code sweep that reruns the auto-formatter in order to make this not be an issue any longer...

Reply:
If that's the case, feel free to leave this set to False, and I'll fix the autostyle issues ina separate PR after this one. Feel free to leave this to False. I'll fix it in a follow up PR.